python_modules/dagster-graphql/dagster_graphql/schema/instigation.py

```diff
@@ -169,87 +171,110 @@ def resolve_logEvents(self, graphene_info: ResolveInfo):
-
-        repository_origin = self._state.origin.external_repository_origin
-        if not graphene_info.context.has_repository_location(
-            repository_origin.repository_location_origin.location_name
-        ):
+        if not graphene_info.context.has_repository_location(self._selector.location_name):
             return None
 
         repository_location = graphene_info.context.get_repository_location(
-            repository_origin.repository_location_origin.location_name
+            self._selector.location_name
         )
-        if not repository_location.has_repository(repository_origin.repository_name):
-            return None
-
-        repository = repository_location.get_repository(repository_origin.repository_name)
-
-        if not repository.has_external_schedule(self._state.name):
+        if not repository_location.has_repository(self._selector.repository_name):
             return None
 
-        external_schedule = repository.get_external_schedule(self._state.name)
-        timezone_str = external_schedule.execution_timezone
-        if not timezone_str:
-            timezone_str = "UTC"
-
-        next_tick_datetime = next(external_schedule.execution_time_iterator(self._timestamp))
-        schedule_time = to_timezone(pendulum.instance(next_tick_datetime), timezone_str)
-        schedule_data: Union[ScheduleExecutionData, SerializableErrorInfo]
-        try:
-            schedule_data = repository_location.get_external_schedule_execution_data(
-                instance=graphene_info.context.instance,
-                repository_handle=repository.handle,
-                schedule_name=external_schedule.name,
-                scheduled_execution_time=schedule_time,
-            )
-        except Exception:
-            schedule_data = serializable_error_info_from_exc_info(sys.exc_info())
+        repository = repository_location.get_repository(self._selector.repository_name)
+
+        if isinstance(self._selector, SensorSelector):
+            if not repository.has_external_sensor(self._selector.sensor_name):
+                return None
+            sensor_data: Union[SensorExecutionData, SerializableErrorInfo]
+            try:
+                sensor_data = repository_location.get_external_sensor_execution_data(
+                    name=self._selector.sensor_name,
+                    instance=graphene_info.context.instance,
+                    repository_handle=repository.handle,
+                    cursor=self._cursor,
+                    last_completion_time=None,
+                    last_run_key=None,
+                )
+            except Exception:
+                sensor_data = serializable_error_info_from_exc_info(sys.exc_info())
+            return GrapheneTickEvaluation(sensor_data)
+        else:
+            if not repository.has_external_schedule(self._selector.schedule_name):
+                return None
+            if not self.timestamp:
+                return None
```

Comment:
these `return None` cases worry me a bit with regards to debugging since there is no indication as to what went wrong and it "fails" silently

Reply:
yea - an artifact of earlier implementation but I don't like it either. I think I'll try and get more coherent errors here.